### PR TITLE
add notes about traffic addressing for waypoint proxies

### LINKED ISSUE #15608: clarify wording for workload-attached waypoints
https://istio.io/latest/docs/ambient/usage/waypoint/#configure-a-pod-to-use-a-specific-waypoint has an unclear description for how traffic reaches a workload-attached waypoint. Add a clarifying point about the original addressing being critical.

### PR DESCRIPTION
**Pull request by ilrudie**
## Description

Fixes #15608 

Adds clarifications for how traffic addressing impacts waypoint selection in the dataplane.

## Reviewers

- [x] Ambient
- [ ] Docs
- [ ] Installation
- [x] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Localization/Translation

### CODE REVIEW

**Review by rinormaloku — APPROVED** [review]
LGTM!

**Comment by kfaseela** [comment]
cc @bleggett @howardjohn

**Review by howardjohn — APPROVED** [review]

File: content/en/docs/ambient/usage/waypoint/index.md [hunk 1]
```diff
@@ -107,10 +107,14 @@ The `--for` parameter to `istioctl waypoint apply` can be used to change the [tr
 | `waypoint-for` value | Traffic type |
```
Comment (bleggett) [inline content/en/docs/ambient/usage/waypoint/index.md:107]:
```suggestion
| `waypoint-for` value |  Destination type |
```

File: content/en/docs/ambient/usage/waypoint/index.md [hunk 2]
```diff
@@ -107,10 +107,14 @@ The `--for` parameter to `istioctl waypoint apply` can be used to change the [tr
...
 | -------------------- | ------------ |
 | `service`            | Kubernetes services |
-| `workload`           | Pod or VM IPs |
+| `workload`           | Pod IPs or VM IPs |
 | `all`                | Both service and workload traffic |
 | `none`               | No traffic (useful for testing) |
 
+{{< tip >}}
```
Comment (bleggett) [inline content/en/docs/ambient/usage/waypoint/index.md:114]: Having 2 tips in the same spot seems unnecessary/duplicative?

Maybe pick one spot for the `tip` (not sure if right here, or below inline is better)

File: content/en/docs/ambient/usage/waypoint/index.md [hunk 3]
```diff
@@ -185,6 +189,10 @@ pod/reviews-v2-5b667bcbf8-spnnh labeled
 
 Any requests from pods in the ambient mesh to the `reviews-v2` pod IP will now be routed through the `reviews-v2-pod-waypoint` waypoint for L7 processing and policy enforcement.
 
+{{< tip >}}
+The original addressing of the traffic is used to determine if a service or workload waypoint should be used. Traffic which is addressed to a service, even though ultimately this would be resolved to a pod IP, is always treated by the ambient mesh as to-service and would use a service-attached waypoint. Workload-attached waypoints are only used when the client addresses traffic to the pod itself.
```
Comment (bleggett) [inline content/en/docs/ambient/usage/waypoint/index.md:193]:
```suggestion
The original destination type of the traffic is used to determine if a service or workload waypoint should be used.
For instance, traffic which is addressed to a service, even though ultimately resolved to a pod IP, is always treated by the ambient mesh as to-service and would use a service-attached waypoint. Waypoints will not be re-transited by the same traffic for the same destination.
```

**Review by dhawton — APPROVED** [review]

**Review by kfaseela — APPROVED** [review]

**Comment by kfaseela** [comment]
Did not see Ben's comments and approved. Hope the dnm label helps

**Comment by ilrudie** [comment]
> Did not see Ben's comments and approved. Hope the dnm label helps

I can take a look and do a follow up PR if needed

File: content/en/docs/ambient/usage/waypoint/index.md (hunk 1 above)
Comment (ilrudie) [reply content/en/docs/ambient/usage/waypoint/index.md:107]: actually more relevant than traffic vs destination here might be adding "original" or something along those lines. Maybe `Original Destination Type`

File: content/en/docs/ambient/usage/waypoint/index.md (hunk 2 above)
Comment (ilrudie) [reply content/en/docs/ambient/usage/waypoint/index.md:114]: They look much closer together in the PR than they actually are in the rendered page. Take a look here and lmk if you still think one or the other needs to go.

https://deploy-preview-15609--preliminary-istio.netlify.app/latest/docs/ambient/usage/waypoint/

Comment (bleggett) [reply content/en/docs/ambient/usage/waypoint/index.md:114]: it still reads a bit as saying the same thing again a few paragraphs later, maybe just hoist the bottom tip right under the table - it makes more sense _to me_ to explicitly call this out once under the types than at the end of the section, but that's just an opinion, so feel free to ignore.

**Comment by ilrudie** [comment]
> > > Did not see Ben's comments and approved. Hope the dnm label helps
> > 
> > 
> > I can take a look and do a follow up PR if needed
> 
> NBD either way, your call.

I like what you suggested for rewordings. On the fence re: 2 or 1 tip...

**Pull request merged**